### PR TITLE
Containerization Services

### PR DESCRIPTION
**Pull request by wxy1203**
Dockerize the services:

- Frontend ✅
- Collaboration Service ✅
- Matching Service ✅
- User Service
- Question Service ✅ 

Docker compose can make containers run as expected.

Problem encountered:
- if sign out, cannot sign back in, don't know where goes wrong

### CODE REVIEW

File: collaboration-service/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
+FROM node:17-alpine
```
Comment (deeyonn) [inline collaboration-service/Dockerfile]: Our project uses node 18 LTS, not sure if downgrading will cause any compatibility issues

File: collaboration-service/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+WORKDIR /app
+
+# Copy package.json files into the container.
+COPY package*.json ./
+
+# Copy .next files into the container.
+COPY .next /app/.next
```
Comment (deeyonn) [inline collaboration-service/Dockerfile:8]: iirc, .next files are regenerated at runtime, so it will be replaced with new ones whenever the app runs

**Review by xiaoying1129 — APPROVED** [review]
lgtm

File: docker-compose.yaml [hunk 3]
```diff
@@ -13,12 +13,18 @@ services:
...
+      - GITHUB_ID=4e29058a74f57c9fd62a
+      - GITHUB_SECRET=18688aafe7221621dc420439cfe1ca753b9fb5c6
+
 
   collaboration-service:
     build: ./collaboration-service
     container_name: collaboration_service_container
     ports:
-      - "3001:3001"
+      - "3001:3000"
```
Comment (deeyonn) [inline docker-compose.yaml:27]: 3000 is already exposed in frontend, not sure if this mapping will cause a conflict of ports with collab service

File: matching-service/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,23 @@
+FROM node:14-alpine
```
Comment (deeyonn) [inline matching-service/Dockerfile]: is there a reason for some services to use specific node 18, 17, 14?

**Review by deeyonn — APPROVED** [review]
I left some comments for review but if there are no changes, it LGTM

File: collaboration-service/Dockerfile (hunk 1 above)
Comment (wxy1203) [reply collaboration-service/Dockerfile]: Okay, use node 18

File: docker-compose.yaml (hunk 3 above)
Comment (wxy1203) [reply docker-compose.yaml:27]: Liveblocks uses 3000 as port, so I have to use 3000 as tcp and our local host for whiteboard is 3001.

File: matching-service/Dockerfile (hunk 4 above)
Comment (wxy1203) [reply matching-service/Dockerfile]: No difference for now.

**Comment by wxy1203** [comment]
Don't merge yet!!

File: collaboration-service/Dockerfile (hunk 2 above)
Comment (wxy1203) [reply collaboration-service/Dockerfile:8]: It has to be there. It's to copy the next folder, not generate.

**Comment by wxy1203** [comment]
Resolves #66

**Pull request merged**